src/server/engine_shard.cc

```diff
@@ -717,8 +723,17 @@ void EngineShard::RetireExpiredAndEvict() {
     ttl_delete_target = kTtlDeleteLimit * double(deleted) / (double(traversed) + 10);
   }
 
-  ssize_t eviction_redline = size_t(max_memory_limit * kRedLimitFactor) / shard_set->size();
+  const size_t max_rss_memory =
+      max_memory_limit * std::max(ServerState::tlocal()->rss_oom_deny_ratio, 0.0);
```

Comment:
if rss_oom_deny_ratio = -1 than max_rss_memory is 0?
I believe we should write
rss_limit = ServerState::tlocal()->rss_oom_deny_ratio > 0? max_memory_limit * ServerState::tlocal()->rss_oom_deny_ratio :  std::numeric_limits<size_t>::max()